src/components/MagicCodeInput.tsx

```diff
@@ -264,72 +262,74 @@ function MagicCodeInput(props) {
-                inputRefs.current[indexToFocus].focus();
-                props.onChangeText(props.value.substring(0, indexToFocus));
+                const formElement = inputRefs.current as HTMLFormElement | null;
+                (formElement?.[indexToFocus] as HTMLInputElement).focus();
+                onChangeTextProp(value.substring(0, indexToFocus));
 
                 return;
             }
 
             // If the currently focused index already has a value, it will delete
             // that value but maintain the focus on the same input.
-            if (numbers[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
+            if (focusedIndex && numbers[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
                 setInput(TEXT_INPUT_EMPTY_STATE);
-                numbers = [...numbers.slice(0, focusedIndex), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex + 1, props.maxLength)];
+                numbers = [...numbers.slice(0, focusedIndex), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex + 1, maxLength)];
                 setEditIndex(focusedIndex);
-                props.onChangeText(composeToString(numbers));
+                onChangeTextProp(composeToString(numbers));
                 return;
             }
 
-            const hasInputs = _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length !== 0;
+            const hasInputs = numbers.filter((n) => ValidationUtils.isNumeric(n)).length !== 0;
 
             // Fill the array with empty characters if there are no inputs.
             if (focusedIndex === 0 && !hasInputs) {
-                numbers = Array(props.maxLength).fill(CONST.MAGIC_CODE_EMPTY_CHAR);
+                numbers = Array(maxLength).fill(CONST.MAGIC_CODE_EMPTY_CHAR);
 
                 // Deletes the value of the previous input and focuses on it.
-            } else if (focusedIndex !== 0) {
-                numbers = [...numbers.slice(0, Math.max(0, focusedIndex - 1)), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex, props.maxLength)];
+            } else if (focusedIndex && focusedIndex !== 0) {
+                numbers = [...numbers.slice(0, Math.max(0, focusedIndex - 1)), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex, maxLength)];
             }
 
-            const newFocusedIndex = Math.max(0, focusedIndex - 1);
+            const newFocusedIndex = Math.max(0, (focusedIndex ?? 0) - 1);
 
             // Saves the input string so that it can compare to the change text
             // event that will be triggered, this is a workaround for mobile that
             // triggers the change text on the event after the key press.
             setInputAndIndex(newFocusedIndex);
-            props.onChangeText(composeToString(numbers));
+            onChangeTextProp(composeToString(numbers));
 
-            if (!_.isUndefined(newFocusedIndex)) {
-                inputRefs.current.focus();
+            if (newFocusedIndex !== undefined) {
```

Comment:
```suggestion
            if (newFocusedIndex) {
```

Reply:
@kubabutkiewicz I guess `newFocusedIndex` can be `0`, so it won't work the same